### ISSUE TITLE
Type Errors when importing the plugin

### ISSUE THREAD

**Issue by Zamiell** [eslint-stylistic/eslint-stylistic#506]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

This plugin does not seem to be compatible with the TSESLint [`config`](https://typescript-eslint.io/packages/typescript-eslint/#config) helper function. This is demonstrated by the following trivial config:

#### eslint.config.mjs

```js
// @ts-check

import ESLintPluginStylistic from "@stylistic/eslint-plugin";
import tseslint from "typescript-eslint";

const myConfig = tseslint.config({
  plugins: {
    "@stylistic": ESLintPluginStylistic, // Type error on this line
  },
});
```

You can suppress the type error using a `@ts-expect-error`, and the plugin still appears to work fine. Thus, this seems to be purely a type issue.

When using flat configs, I believe usage of the TSESLint `config` function is best practice in the ecosystem. When using other plugins with the `config` function, such as [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc), [`eslint-plugin-n`](https://github.com/eslint-community/eslint-plugin-n), [`eslint-plugin-unicorn`](https://github.com/sindresorhus/eslint-plugin-unicorn), [`eslint-plugin-import-x`](https://github.com/antfu/eslint-plugin-import-x), there is no such type error, so it must be something wrong with this plugin specifically.

Thus, this should be a high priority to fix, as it affects UX and will cause people to think that they are doing something wrong, when they are really doing nothing wrong.

I think that this is similar (but different) from #481.

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by antfu** [comment]
Can you create a minimal reproduction? It might be fixed by ts-eslint v8.1 with https://github.com/typescript-eslint/typescript-eslint/pull/9657

**Comment by Zamiell** [comment]
Sorry. It doesn't reproduce with a default tsconfig.json. However it does reproduce when we use the recommended Node.js tsconfig.json. Here's a minimal reproduction: https://github.com/Zamiell/style-bug

```
git clone git@github.com:Zamiell/style-bug.git
cd style-bug
npx tsc --noEmit
```

Note that this is the only ESLint plugin that I have seen that has this specific bug.

**Comment by Zamiell** [comment]
@9romise  can you link the commit where it was completed?

**Comment by 9romise** [comment]
> [@9romise](https://github.com/9romise) can you link the commit where it was completed?

I tried in your repo and found that the issue has been fixed since v4.0.0-beta.1

The only one commit is [migrate to ESM](https://github.com/eslint-stylistic/eslint-stylistic/pull/670/files#diff-f63577ff8d0a221af793f7a52cfd80be2f1b1cb92e95e0e31c55dc32a33a40b0). I haven't looked into it deeply, but feel free to bring up anything you find.
